File: Programming-Basics-With-JavaScript/Homeworks-And-Labs/06.Nested-Loop/02-exercise/03.sumPrimeNonPrime.js
function sumPrimeNonPrime(input) {

    let index = 0;
    let current = input[index++]

    let sumOfAllPrime = 0;
    let sumOfAllNonPrime = 0;

    while (current !== "stop") {
        let currentNum = Number(current);
        if (currentNum < 0) {
            console.log("Number is negative.");
        } else {
            let isPrime = true;
            for (let i = 2; i <= currentNum - 1; i++) {
                if (currentNum % i === 0) {
                    isPrime = false;
                    break;
                }
            }
            if (isPrime) {
                sumOfAllPrime += currentNum;
            } else {
                sumOfAllNonPrime += currentNum;
            }
        }

        current = input[index++];
    }

    console.log(`Sum of all prime numbers is: ${sumOfAllPrime}`);
    console.log(`Sum of all non prime numbers is: ${sumOfAllNonPrime}`);
}
sumPrimeNonPrime(["3",
    "9",
    "0",
    "7",
    "19",
    "4",
    "stop"]);

File: Python_Basics/Homeworks-And-Labs/06.nested_loops/02.exercise/03.sum_prime_non_prime.py
current = input()

sum_of_all_prime = 0
sum_of_all_non_prime = 0

while current != "stop":
    current_num = int(current)
    if current_num < 0:
        print("Number is negative.")
    else:
        is_prime = True
        for i in range(2, current_num):
            if current_num % i == 0:
                is_prime = False
                break
                
        if is_prime:
            sum_of_all_prime += current_num
        else:
            sum_of_all_non_prime += current_num
            
    current = input()
    
print(f'Sum of all prime numbers is: {sum_of_all_prime}')
print(f'Sum of all non prime numbers is: {sum_of_all_non_prime}')
